hail/src/main/scala/is/hail/expr/ir/BlockMatrixIR.scala

```diff
@@ -149,7 +149,7 @@ class BlockMatrixLiteral(value: BlockMatrix) extends BlockMatrixIR {
   val blockCostIsLinear: Boolean = true // not guaranteed
 }
 
-case class BlockMatrixMap(child: BlockMatrixIR, eltName: String, f: IR) extends BlockMatrixIR {
+case class BlockMatrixMap(child: BlockMatrixIR, eltName: String, f: IR, needsDense: Boolean) extends BlockMatrixIR {
```

Comment:
If we're manually inserting densifies before the map in python, why do we need the `needsDense` flag on `BlockMatrixMap` in the IR?

Reply:
We don't strictly need it, and I'm happy to take it out; it will throw a compile-time/parse-time error if `needsDense` is true but the child is sparse. I think we can avoid this by just making sure never to have that be true, though.

Reply:
Actually, I think it's good the way it is. Thanks for explaining.